go/kex2/provisioner.go

```diff
@@ -162,6 +173,47 @@ func (p *provisioner) runProtocolWithCancel() (err error) {
 }
 
 func (p *provisioner) runProtocol() (err error) {
+	var fallback bool
+	p.debug("+ provisioner#runProtocol: try V2")
+	fallback, err = p.runProtocolV2()
+	p.debug("- provisioner#runProtocol -> %v, %v", fallback, err)
+	if fallback {
+		p.debug("+ provisioner#runProtocol: fallback to V1")
+		err = p.runProtocolV1()
+		p.debug("- provisioner#runProtocol V1 -> %v", err)
+	}
+	return err
+}
+
+func (p *provisioner) runProtocolV2() (fallback bool, err error) {
+	cli := keybase1.Kex2Provisionee2Client{Cli: rpc.NewClient(p.xp, nil)}
+	var helloArg keybase1.Hello2Arg
+	helloArg, err = p.arg.Provisioner.GetHello2Arg()
+	if err != nil {
+		return false, err
+	}
+	var res keybase1.Hello2Res
+	if res, err = cli.Hello2(context.TODO(), helloArg); err != nil {
+		if strings.Contains(err.Error(), "protocol not found: keybase.1.Kex2Provisionee2") {
```

Comment:
this is slightly ugly, but not much we can do subject to backwards compatibility here.